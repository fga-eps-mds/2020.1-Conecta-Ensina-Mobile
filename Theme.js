import React from 'react';
import {ThemeProvider} from 'styled-components';

const theme = {
  colors: {
    fundoAzul: '#184E79',
    cinzaClaro: '#D9D9D9',
    cinzaEscuro: '#596C7C',
    vermelho: '#E46270',
    azulClaro: '#297FB8',
    verde: '#3FB93B',
    verdeEscuro: '#2D9D49',
    botÃ£oLoginNaoApertado: '#8B99A5',
    branco: '#ffffff',
    preto: 'black',
  },
  font: 'Roboto',
  fontSizes: {
    smaller: '12px',
    small: '14px',
    bigSmall: '16px',
    smallMedium: '20px',
    medium: '25px',
    bigMedium: '30px',
    big: '45px',
  },
};

const Theme = ({children}) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
