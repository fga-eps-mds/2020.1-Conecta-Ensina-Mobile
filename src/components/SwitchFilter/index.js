import React from 'react';
import SwitchSelector from 'react-native-switch-selector';
import {Container} from './styles';
import {withTheme} from 'styled-components/native';

function SwitchFilter({theme, onChange}) {
  const estiloOptions = [
    {label: 'Online', value: 'Online'},
    {label: 'Presencial', value: 'Presencial'},
  ];
  return (
    <Container>
      <SwitchSelector
        options={estiloOptions}
        initial={0}
        value={0}
        onPress={(value) => onChange(value)}
        textColor={theme.colors.preto}
        selectedColor={theme.colors.branco}
        buttonColor={theme.colors.vermelho}
        borderRadius={10}
        buttonMargin={3}
        height={43}
        fontSize={12}
      />
    </Container>
  );
}

export default withTheme(SwitchFilter);
