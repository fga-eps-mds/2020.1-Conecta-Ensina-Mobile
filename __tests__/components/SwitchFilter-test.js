import React from 'react';
import 'react-native';
import {render, fireEvent} from '@testing-library/react-native';
import SwitchFilter from '../../src/components/SwitchFilter';
import Theme from '../../Theme';

describe('Testing SwitchFilter Component', () => {
  /*  const onChange = jest.fn();

  test('SwitchSelector should be pressed', () => {
    const {getByTestId, UNSAFE_getByType} = render(
      <Theme>
        <SwitchFilter onChange={onChange} />
      </Theme>,
    );

    const switchSelector = UNSAFE_getByType(SwitchFilter);

    fireEvent.press(switchSelector);

    expect(onChange).toBeCalled();
  });
*/
  test('Component should be render', () => {
    const tree = render(
      <Theme>
        <SwitchFilter />
      </Theme>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
