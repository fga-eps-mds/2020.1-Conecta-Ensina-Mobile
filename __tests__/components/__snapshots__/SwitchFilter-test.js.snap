// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SwitchFilter Component Component should be render 1`] = `
<View
  style={
    Array [
      Object {
        "width": 190,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "height": 49,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#C9C9C9",
              "borderRadius": 10,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#ffffff",
                    "fontSize": 12,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  Object {},
                ]
              }
            >
              Online
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "black",
                    "fontSize": 12,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  Object {},
                ]
              }
            >
              Presencial
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
